library/core/src/ptr/non_null.rs

```diff
@@ -412,6 +413,7 @@ impl<T: ?Sized> NonNull<T> {
     #[rustc_const_unstable(feature = "const_ptr_as_ref", issue = "91822")]
     #[must_use]
     #[inline(always)]
+    #[kani::requires(ub_checks::can_dereference(self.as_ptr() as *const()))]
```

Comment:
Do you need to cast the `self.as_ptr()` result to a *const here?

Reply:
That's a good point. That's because the generic type (T) in this API is no strictly Sized, so we need to convert it into  *const as a workaround, related discussion: https://github.com/model-checking/verify-rust-std/discussions/141